packages/extension-api/src/extension-api.d.ts

```diff
@@ -423,6 +423,10 @@ declare module '@podman-desktop/api' {
     export const onDidUnregisterContainerConnection: Event<UnregisterContainerConnectionEvent>;
     export const onDidRegisterContainerConnection: Event<RegisterContainerConnectionEvent>;
     export function getContainerConnections(): ProviderContainerConnection[];
+    export function getMatchingProviderLifecycleContext(
+      providerId: string,
+      providerConnectionInfo: ProviderContainerConnectionInfo | ProviderKubernetesConnectionInfo,
+    ): LifecycleContextImpl;
```

Comment:
I don't see how we can return a LifecycleContextImpl object
it's an internal implementation